ha/util/consul_kv_store.py

```diff
@@ -103,10 +116,20 @@ def key_exists(self, key: str):
         Return:
             bool: True if key exists else False.
         """
-        _, data = self._consul.kv.get(self._prepare_key(key), recurse=True)
-        if data is None:
-            return False
-        return True
+        k = self._prepare_key(key)
+
+        # First find in cache if batch put is enabled
+        Log.debug(f"[consul-op] is key exist in cache: {k}")
+        if self._enable_batch_put and [key for key in self._payload if key.startswith(k)]:
+            Log.debug(f"[consul-op] key {k} found in cache: True")
+            return True
```

Comment:
Here we return True if key exists (i.e. without checking if data exists or not ) and when looking up in consul we return True only if data exists.
Why this difference in behavior ?

Reply:
Thinking on this will reply after checking with sample program

Reply:
This check should be on key because:

data return from consul.kv.get() is not a only value but many parameters.
so for the key even if we set None in consul the data returns many value.
I set 1 key 'fake_events' with None value and debug what data contains and found below the object
```
[{'LockIndex': 0, 'Key': 'cortx>ha>v2>fake_events', 'Flags': 0, 'Value': None, 'CreateIndex': 1355997, 'ModifyIndex': 1355997}]
```